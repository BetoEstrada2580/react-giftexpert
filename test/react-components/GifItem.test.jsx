import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Test de <GifItem/>', () => {

    const title = "Berserk";
    const url = "https://i.blogs.es/755972/berserk-manga-2/1366_2000.jpeg";

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(
            <GifItem
                title={title}
                url={url}
            />);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect(src).toEqual(url);
        expect(alt).toEqual(title);
    })

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })

});