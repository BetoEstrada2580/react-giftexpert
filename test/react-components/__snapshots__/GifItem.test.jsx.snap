// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Berserk"
      src="https://i.blogs.es/755972/berserk-manga-2/1366_2000.jpeg"
    />
    <p>
      Berserk
    </p>
  </div>
</div>
`;
